runtime/port/sysvipc/j9sharedhelper.c

```diff
@@ -164,13 +164,26 @@ createDirectory(struct J9PortLibrary *portLibrary, char *pathname, uintptr_t per
 			/* Change permission on new dir, note that for last dir in path this will be changed in ensureDirectory */
 			rc = changeDirectoryPermission(portLibrary, tempPath, permission);
```

Comment:
This is unreasonable, and will just fail where the process does not own the relevant directory. Even if we have permission to change the permissions, I don't think we should do that.

Reply:
No, it happens for each directory along the way: `/home`, then `/home/keithc`, then `/home/keithc/tmp`, etc. I don't have permission to change `/home`, for example. You'll notice `rc` was not checked. The caller should set the permission of the final directory (I believe it does).

Reply:
I focused on the `tempPath` argument which could also be any (non-root) prefix of `pathname`, but I think you're right, that call should only apply to newly created directories.